bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/plugin/ClangdPlugin.java

```diff
@@ -56,7 +56,11 @@ public void start(BundleContext context) throws Exception {
 		compileCommandsMonitor = new CompileCommandsMonitor(workspace).start();
 		cProjectChangeMonitor = new CProjectChangeMonitor().start();
 		configFileMonitor = new ClangdConfigFileMonitor(workspace).start();
-		formatMonitor = new ClangFormatMonitor(new ClangFormatUtils()).start();
+
+		ServiceTracker<ClangFormatFile, ClangFormatFile> clangFormatTracker = new ServiceTracker<>(context,
```

Comment:
`ServiceTracker` usage looks excessive for obtaining service, and nobody closes it.